File: 2.linked-list/single-linked-list/delete-node/delete_node.rb
def delete_node(head,key)
  while (head && head.next && head.val == key) do
    head = head.next
  end
  if (!head.nil?)
    return head
  end
  prev = head
  current = head.next
  while current do
    if current.val == key
      prev.next = current.next
    else
      prev = current
    end
    current = current.next
  end
  prev.next = null
  return head
end

File: 2.linked-list/single-linked-list/delete-node/deleteNode.js
let deleteNode = function (head, key) {
  while ((head.val === key) & head & head.next) {
    head = head.next;
  }
  if (!head) {
    return head;
  }

  let prev = head;
  let current = head.next;
  while (current) {
    if (current.val !== key) {
      prev.next = current.next;
    } else {
      prev = current;
    }
    current = current.next; // new current
  }
  prev.next = null;
  return head;
};
